### ISSUE TITLE
Matplotlib lessons in PyCharm

### ISSUE THREAD

**Issue by donkirkby** [donkirkby/live-py-plugin#176]
After #52 adds support for displaying a live matplotlib plot, I'd like to add a set of matplotlib lessons. The minimal version would be to include the lesson text in the comments of a Python file, along with a marker comment that points to the goal file. Then PyCharm would display both the current plot and the goal plot, and maybe an option to display the image diff. (Image diff when the mouse hovers over?)

A future issue could add some navigation between the lessons and a display of how many you've completed. Maybe resize the plots to fit in the window?

* [x] display goal plot as well
* [x] display single plot when they match, or replace the goal with a big green check mark
* [x] display diff (on hover?)
* [x] resize the plots to fit (separate issue: #183)
* [x] new issue for navigation and progress tracking: #184

Here's a trivial example lesson:

    # format_lesson.py
    """ Changing line colours

    You can change the colour of a line by passing a colour abbreviation in the format string
    of the plot() method. Add a format string to these two plot() calls to make the heights
    red and the weights green.
    :lesson goal file: format_goal.py
    """
    from matplotlib import pyplot as plt
    
    heights = [180, 180, 180, 173]
    weights = [72, 72, 64, 62]
    
    plt.plot(heights)
    plt.plot(weights)
    plt.show()

The goal file would just add the format strings:

    # format_goal.py
    from matplotlib import pyplot as plt
    
    heights = [180, 180, 180, 173]
    weights = [72, 72, 64, 62]
    
    plt.plot(heights, 'r')
    plt.plot(weights, 'g')
    plt.show()
